server/stt/stt_events_ml.py

```diff
@@ -280,18 +280,29 @@ def set_location_details(self, item, location_xml, notes):
         locations_service = get_resource_service("locations")
         if locations_service is not None:
             try:
-                existing_location = locations_service.find_one(
-                    req=None,
-                    name=location.get("name"),
-                    **{"address.city": location.get("address", {}).get("city")},
+                stt_id = (
+                    location.get("address", {}).get("extra", {}).get("sttlocationalias")
                 )
+                custom_guid = f"urn:stt:location:{stt_id}" if stt_id else None
```

Comment:
we would need a guid, so if there is no stt_id I wouldn't store it probably and just keep it on the event